### PR TITLE
Function Calling with Hugging Face

### PR DESCRIPTION
**Pull request by Jofthomas**
Blog Post on how to do Function call with Hugging Face models

To merge:
Wednesday at 6.30 pm

### CODE REVIEW

**Comment by Jofthomas** [comment]
Second PR @philschmid @andrewrreed 

hey @drbh I added you here as author as said previously.

File: Function_Calling.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+- user: jofthomas
+- user: drbh
+- user: kkondratenko
+  guest: true
+---
+# Function Calling in Hugging Face is here !
+
+## Introduction
+
+A few weeks ago, we introduced to you the brand new [Messages AP](https://huggingface.co/blog/tgi-messages-api)I that provided OpenAI compatibility with Text Generation Inference (TGI) and Inference Endpoints.
```
Comment (linda-yeah) [inline Function_Calling.md:14]: delete the "I"

Comment (linda-yeah) [reply Function_Calling.md:14]: also AP --> API

File: Function_Calling.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+We wrote at the time that “*The Messages API does not currently support function calling” this is a limitation that has now been lifted !*
+
+Starting with version **1.4.5,** TGI offers an API compatible with the OpenAI Chat Completion API with the addition of the `tools` and the `tools_choice` keys. This change as been propagated in the**`huggingface_hub`** version **0.23.0.** Meaning any Huggingface endpoint can now call some tools if using a newer version than those.
+
+This new feature is available in Inference Endpoints (dedicated and serverless). and we’ll showcase how you can start building your open-source agents right away.
+
+To get you started quickly, we’ve included detailed code examples of how to:
+
+- Create an Inference Endpoint
+- Call Function with the InferencClient
```
Comment (linda-yeah) [inline Function_Calling.md:25]: typo: InferenceClient

Comment (linda-yeah) [reply Function_Calling.md:25]: also another typo: "This change has been propogated in the **`huggingface_hub`** version **0.23.0.**, meaning any Hugging Face endpoint can now call some tools if using a newer version."

File: Function_Calling.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+- Complete control and transparency over models and data
+- No more worrying about rate limits
+- The ability to fully customize systems according to your specific needs
+
+Lets see how.
+
+### With the InferencClient from Hugging Face
+
+Function can directly be called with the serverless API, or with any endpoint by with the enpoint url.
```
Comment (linda-yeah) [inline Function_Calling.md:101]: typo: endpoint URL

File: Function_Calling.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+# pause our running endpoint
+endpoint.pause()
+
+# optionally delete
+endpoint.delete()
+```
+
+## Conclusion
+
+You can now call some tools with Hugging Face models in the different frameworks. We can’t wait to see what use cases you will power with open LLMs and tools !
```
Comment (linda-yeah) [inline Function_Calling.md:363]: add a call to action with a link to either docs or place where users can try this out for themselves

**Review by efriis — COMMENTED** [review]
Should we rename to "Tool Calling" to match the general direction of what this feature is called on other platforms?

Will read through deeper later this afternoon

**Comment by Jofthomas** [comment]
Would make sense indeed.

**Pull request closed without merging**